templates/cicd/pipeline_cfn.yml

```diff
@@ -7,7 +7,20 @@
 # limitations under the License.
 AWSTemplateFormatVersion: '2010-09-09'
 Description: CodePipeline for {{$.AppName}}
+Parameters:
+  ProviderType:
+    Type: String
+    Default: {{.Source.ProviderName}}
+Conditions:
+  IsBitbucket:
+    !Equals [!Ref ProviderType, "Bitbucket"]
 Resources:
+  SourceConnection:
+    Type: AWS::CodeStarConnections::Connection
+    Condition: IsBitbucket
+    Properties:
+      ConnectionName: Copilot-Connection
```

Comment:
nit: can we name the connection based on the repo name and provider?
```yaml
ConnectionName: copilot-{{.Source.ProviderName}}-{{.Source.Repository}} 
```
(Or something like this) this way when we add more options besides bitbucket, the customer can easily identify which connection is for which repo

Reply:
Good call! 
There's a 32-character max length and "Provider" is a column in the console table, so I'm making it `copilot-{{.Source.Repository}}.

Reply:
Taking reference to connection name out of `pipeline_update` for now.

Reply:
[Discussed offline. Adding method to generate ConnectionName.}